File: src/test/java/TestDivision.java
import org.junit.Assert;
import org.junit.Test;


public class TestDivision {

    @Test
    public void testDividedBySimple() {
        var fraction = TestUtils.createFraction(12, 7);
        var result = fraction.dividedBy(new NotImplementedFraction(6, 11));

        Assert.assertEquals((Integer) 22, result.getNumerator());
        Assert.assertEquals((Integer) 7, result.getDenominator());
    }

    @Test
    public void testDividedByNormalisation() {
        var fraction = TestUtils.createFraction(4, 6);
        var result = fraction.dividedBy(new NotImplementedFraction(7, 5));

        Assert.assertEquals((Integer) 10, result.getNumerator());
        Assert.assertEquals((Integer) 21, result.getDenominator());
    }

}

File: src/test/kotlin/TestDivision.kt
import org.junit.Assert
import org.junit.Test

class TestDivision {
    @Test
    fun testDividedBySimple() {
        val fraction = TestUtils.createFraction(4, 1)
        val result = fraction / NotImplementedFraction(3, 1)
        Assert.assertEquals(4, result.numerator)
        Assert.assertEquals(3, result.denominator)
    }

    @Test
    fun testDividedByNormalisation() {
        val fraction = TestUtils.createFraction(4, 6)
        val result = fraction / NotImplementedFraction(7, 5)
        Assert.assertEquals(10, result.numerator)
        Assert.assertEquals(21, result.denominator)
    }
}